src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIBasic.java

```diff
@@ -1240,15 +1240,10 @@ public boolean checkIfRequestForItemExistOrCreateAsynch(@NotNull final ItemStack
      */
     public boolean tryTransferFromPosToWorker(final BlockPos pos, @NotNull final Predicate<ItemStack> predicate)
     {
-        final TileEntity entity = world.getTileEntity(pos);
-        if (entity.hasCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, EnumFacing.DOWN))
-        {
-            return InventoryUtils.transferXOfFirstSlotInItemHandlerWithIntoNextFreeSlotInItemHandler(
-                    new InvWrapper((IInventory) entity),
-                    predicate,
-                    Constants.STACKSIZE,
-                    new InvWrapper(worker.getInventoryCitizen()));
-        }
-        return false;
+        return InventoryUtils.transferXOfFirstSlotInProviderWithIntoNextFreeSlotInItemHandler(
+                (ICapabilityProvider) world.getTileEntity(pos),
```

Comment:
How are we guaranteeing that we can convert these two? (are tile entities always ICapabilityProviders?)

Reply:
Yeah